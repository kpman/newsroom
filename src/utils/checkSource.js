const invariant = require('invariant');

module.exports = (source, sourcesInfo) => {
  const result = sourcesInfo.some(sourceInfo => sourceInfo.title === source);
  invariant(result, `The source \`${source}\` is not in the sources(.opml)!`);
};
