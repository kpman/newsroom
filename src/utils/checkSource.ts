import invariant from 'invariant';

export default (source, sourcesInfo) => {
  const result = sourcesInfo.some((sourceInfo) => sourceInfo.title === source);
  invariant(result, `The source \`${source}\` is not in the sources(.opml)!`);
};
