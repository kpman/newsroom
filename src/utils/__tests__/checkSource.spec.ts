import checkSource from '../checkSource';

it('should be defined', () => {
  expect(checkSource).toBeDefined();
});

describe('#checkSource', () => {
  it('should check the source in the sources', () => {
    const source = 'fb';
    const sources = [
      {
        title: 'fb',
      },
    ];
    expect(() => checkSource(source, sources)).not.toThrow();
  });

  it('should throw error', () => {
    const source = 'fb';
    const sources = [
      {
        title: 'snap',
      },
    ];
    expect(() => checkSource(source, sources)).toThrowErrorMatchingSnapshot();
  });
});
