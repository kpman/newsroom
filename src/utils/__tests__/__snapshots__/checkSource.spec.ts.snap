// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#checkSource should throw error 1`] = `"The source \`fb\` is not in the sources(.opml)!"`;
